crates/sparrow-runtime/src/metadata/raw_metadata.rs

```diff
@@ -1,13 +1,33 @@
-use std::fs::File;
 use std::io::{BufReader, Cursor};
+use std::str::FromStr;
 use std::sync::Arc;
 
 use arrow::datatypes::{DataType, Field, Schema, SchemaRef};
+use error_stack::{IntoReport, IntoReportCompat, ResultExt};
 use parquet::arrow::arrow_reader::ParquetRecordBatchReaderBuilder;
 use sparrow_api::kaskada::v1alpha::file_path::Path;
+use tempfile::NamedTempFile;
 use tracing::info;
 
 use crate::metadata::file_from_path;
+use crate::object_store_url::ObjectStoreKey;
+use crate::{ObjectStoreRegistry, ObjectStoreUrl};
+
+#[derive(derive_more::Display, Debug)]
+pub enum Error {
+    #[display(fmt = "object store error for path: {_0}")]
+    ObjectStore(String),
+    #[display(fmt = "no object store registry")]
+    NoObjectStoreRegistry,
+    #[display(fmt = "local file error")]
+    LocalFile,
+    #[display(fmt = "download error")]
+    DownloadIssue,
+    #[display(fmt = "reading schema error")]
+    ReadSchemaIssue,
```

Comment:
nit: Suffixing these with `Issue` is just adding a synonym for `Error`. I would leave off, and prefer `Error::Download { context: String }` etc, probably. Or just attach context manually if preferred.